ansible_base/lib/utils/models.py

```diff
@@ -101,7 +101,12 @@ def get_system_user() -> Optional[AbstractUser]:
                 )
             )
         )
-        system_user = create_system_user(user_model=get_user_model())
+        try:
+            # There are issues in the tests where, during teardown the system user can not be created
```

Comment:
Because of a broken transaction?

Reply:
I haven't been able to determine why this happening. I thought pytest was run within a transaction but it for some reason the resource_registry signals are being processed like there has been a reverse migration. I tried adding code to `initialize_resources` in `ansible_base.resource_registry.apps` to detect a reverse migration but it didn't help. This method eventually does `ResourceType.objects.update_or_create` which will then try and create the system user. But things are "missing" at this point causing the exception we are catching above. But this seems like can rear its head in multiple exception types depending on the test which is why I just catch Exception.